tests/performance/injective_functions_inside_uniq.xml

```diff
@@ -0,0 +1,7 @@
+<test>
+
+    <query>SELECT uniq(negate(negate(negate(number)))) FROM numbers(500000000)</query>
```

Comment:
A test on `analyze` might be more appropriate:

```
:) set enable_debug_queries = 1
:) analyze select count(*) from (select 1 x) group by hex(x);

┌─explain──────────────────────────────────────────┐
│ SELECT count()
FROM 
(
    SELECT 1 AS x
)
GROUP BY x │
└──────────────────────────────────────────────────┘
```

With `analyze` query, you can directly check that your query was rewritten as intended, instead of guessing it from the performance (it's expensive). You can add this test as a `.sql` test to `tests/queries/0_stateless`.